core/targets.py

```diff
@@ -105,17 +105,21 @@ def analysis(targets, check_ranges, check_subdomains, subs_temp, range_temp, log
         elif target_type(target) == 'SINGLE_IPv6':
             yield target
 
+        
         elif target_type(target) == 'RANGE_IPv4' or target_type(target) == 'CIDR_IPv4':
-            IPs = IPRange(target, range_temp, language)
-            if not enumerate_flag:
-                info(messages(language, "checking").format(target))
-            if type(IPs) == netaddr.ip.IPNetwork:
-                for IPm in IPs:
-                    yield IPm
-            elif type(IPs) == list:
-                for IPm in IPs:
-                    for IP in IPm:
-                        yield IP
+            if("shodan_scan" in scan_method and "/" in target):
```

Comment:
do not modify the core for one module - this creates a hard link between the core and shodan module - the modules should be independent